Update developer guide NFRs (#44)
AY2021S2-CS2103T-T11-1/tp: docs/DeveloperGuide.md

*{More to be added}*

### Non-Functional Requirements  
* Technical requirements:  
    * TutorBuddy should work on both 32-bit and 64-bit environments.
    * TutorBuddy should work on any _mainstream OS_ with Java `11` or above installed.
    * The user should have enough memory on their computer to ensure that data will be stored in the application without errors.
* Performance requirements:
    * TutorBuddy should be able to hold up to _500 students_ or _500 sessions_ without a noticeable dip in performance for typical usage.
    * Response time from any command should be within 1 second.
* Constraints:
    * TutorBuddy should be backward compatible with data produced by earlier versions of the system.
* Quality requirements:

>     * The user should take less than 1 hours to learn the different functionality that TutorBuddy has from the user guides.

Reviewer: Slight change in phrasing: 
```suggestion
    * The user should take no longer than 1 hour to learn the different functionalities of TutorBuddy from the user guide.
```

* Process requirements:
    * TutorBuddy should be completed before AY20/21 Week 13
* Any other noteworthy points:

>     * A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.

Reviewer: Maybe can change it to like that to be more specific?
```suggestion
    * A user with above-average typing speed for regular English text (i.e. not code, not system admin commands) should be able to use most of the functionality in TutorBuddy faster using commands than using the mouse.
```
